processor/src/main/java/org/mapstruct/ap/spi/DefaultAccessorNamingStrategy.java

```diff
@@ -90,6 +90,7 @@ protected boolean isBuilderSetter(ExecutableElement method) {
         return method.getParameters().size() == 1 &&
             !JAVA_JAVAX_PACKAGE.matcher( method.getEnclosingElement().asType().toString() ).matches() &&
             !isAdderWithUpperCase4thCharacter( method ) &&
+            //typeUtils.isAssignable( method.getReturnType(), method.getEnclosingElement().asType() )
```

Comment:
have you tried `typeUtils#asMemberOf`? If you have a return type or method parameter part of a containing executable-type or containing class you can evaluate the generic arguments as member of this containing type...

Reply:
Or.. we should not give a generic `ExecutableElement` but a fully resolved one to this method. Don't know if that's possible as well..

Reply:
That might actually work. Although I am not sure what it might mean from a performance point of view. As we would need to resolve all methods, even though they might never be used.

Reply:
I added some comments to #1253